spec/ruby/shared/signal/trap.rb

```diff
@@ -82,6 +82,14 @@
       ScratchPad.recorded.should == :callable_trap
     end
 
+    it "raises an exception for a non-callable at the point of use" do
+      not_calllable = Object.new
+      @object.trap(:HUP, not_calllable)
+      -> {
+        Process.kill :HUP, Process.pid
```

Comment:
We need something like `Thread.pass while true` below, as there can be some time between the signal being sent, being received, and being processed on the main thread.